cmd/ttn-lw-cli/commands/applications_pubsub.go

```diff
@@ -180,14 +201,51 @@ var (
 			}
 			pubsub.ApplicationPubSubIdentifiers = *pubsubID
 
-			if nats, _ := cmd.Flags().GetBool("nats"); nats {
-				serverURL, _ := cmd.Flags().GetString("nats-server-url")
-				pubsub.Provider = &ttnpb.ApplicationPubSub_NATS{
-					NATS: &ttnpb.ApplicationPubSub_NATSProvider{
-						ServerURL: serverURL,
+			for name, p := range map[string]struct {
+				provider ttnpb.ApplicationPubSub_Provider
+				flags    *pflag.FlagSet
+				loadData func() error
+			}{
+				"nats": {
+					provider: &ttnpb.ApplicationPubSub_NATS{},
+					flags:    natsProviderApplicationPubSubFlags,
+					loadData: func() error { return nil },
+				},
+				"mqtt": {
+					provider: &ttnpb.ApplicationPubSub_MQTT{},
+					flags:    mqttProviderApplicationPubSubFlags,
+					loadData: func() error {
+						if useTLS, _ := cmd.Flags().GetBool("mqtt.use-tls"); useTLS {
+							for _, name := range []string{
+								"mqtt.tls-ca",
+								"mqtt.tls-client-cert",
+								"mqtt.tls-client-key",
+							} {
+								data, err := getDataBytes(name, cmd.Flags())
+								if err != nil {
+									return err
+								}
+								err = cmd.Flags().Set(name, hex.EncodeToString(data))
```

Comment:
Why setting the hex value of the certs as the flag values?

Reply:
`util.SetFields` expects `bytes` proto fields to be provided as `hex` values.